arcgis-ios-sdk-samples/Geometry/Spatial operations/SpatialOperationsViewController.swift

```diff
@@ -15,146 +15,159 @@
+        resultGraphic = graphic
         
-        //create result graphic if not present
-        if self.resultGraphic == nil {
-            let graphic = AGSGraphic(geometry: resultGeometry, symbol: symbol, attributes: nil)
-            self.graphicsOverlay.graphics.add(graphic)
-            self.resultGraphic = graphic
-        }
-            //update the geometry if already present
-        else {
-            self.resultGraphic.geometry = resultGeometry
-        }
+        // An overlay to display polygon graphics.
+        let graphicsOverlay = AGSGraphicsOverlay()
+        
+        // Add graphics to graphics overlay.
+        graphicsOverlay.graphics.addObjects(from: [polygon1Graphic, polygon2Graphic, graphic])
+        return graphicsOverlay
     }
     
-    // MARK: - Navigation
-    
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        if segue.identifier == "OperationsListSegue" {
-            let controller = segue.destination as! OperationsListViewController
-            controller.delegate = self
-            controller.operations = self.operations
-            controller.presentationController?.delegate = self
-            controller.preferredContentSize = CGSize(width: 200, height: 200)
+    private func performOperation(_ operation: SpatialOperation) {
+        let resultGeometry: AGSGeometry?
+        switch operation {
+        case .none:
+            resultGeometry = nil
+        case .union:
+            resultGeometry = AGSGeometryEngine.union(ofGeometry1: polygon1, geometry2: polygon2)!
+        case .difference:
+            resultGeometry = AGSGeometryEngine.difference(ofGeometry1: polygon1, geometry2: polygon2)!
+        case .symmetricDifference:
+            resultGeometry = AGSGeometryEngine.symmetricDifference(ofGeometry1: polygon1, geometry2: polygon2)!
+        case .intersection:
+            resultGeometry = AGSGeometryEngine.intersection(ofGeometry1: polygon1, geometry2: polygon2)!
         }
+        // Update the geometry.
+        resultGraphic.geometry = resultGeometry
     }
     
-    // MARK: - OperationsListVCDelegate
-    
-    func operationsListViewController(_ operationsListViewController: OperationsListViewController, didSelectOperation index: Int) {
-        if index == 0 {
-            if let resultGraphic = resultGraphic {
-                self.graphicsOverlay.graphics.remove(resultGraphic)
-                self.resultGraphic = nil
+    @IBAction func chooseOperationBarButtonTapped(_ sender: UIBarButtonItem) {
+        let selectedIndex = SpatialOperation.allCases.firstIndex(where: { $0 == selectedOperation })
+        
+        let controller = OptionsTableViewController(labels: SpatialOperation.allCases.map { $0.label }, selectedIndex: selectedIndex) { [weak self] newIndex in
+            guard let self = self else { return }
+            let newOperation = SpatialOperation.allCases[newIndex]
+            if self.selectedOperation != newOperation {
```

Comment:
This check is not needed due to [recent changes](https://github.com/Esri/arcgis-runtime-samples-ios/blob/6cbe5b992cd91a6596fe9c0e299854170c947e15/arcgis-ios-sdk-samples/Content%20Display%20Logic/Controllers/OptionsTableViewController.swift#L90) to `OptionsTableViewController`. The `onChange` callback is now only called when the selection actually changes.

Reply:
Aha that's nice! 🤩